src/native/ceph_s3select/ceph_s3lect_napi.cpp

```diff
@@ -0,0 +1,428 @@
+            } else {
+                rc = csv_object->run_s3select_on_stream(this->select, this->buffer, this->buffer_len, SIZE_MAX);
+            }
+        } else {//currenty only json
+            if (is_flush) {
+                rc = json_object->run_s3select_on_stream(this->select, this->buffer, 0, 0);
+            } else {
+                rc = json_object->run_s3select_on_stream(this->select, this->buffer, this->buffer_len, SIZE_MAX);
+            }
+        }
+	    if (rc < 0) {
+            throw Napi::Error::New(Env(), XSTR() << "s3select: csv.run_s3select_on_stream failed " << csv_object->get_error_description());
+	    }
+
+        //std::cout << "res.len = " << aws.get_result().length() << " payload line len = " << strlen(PAYLOAD_LINE) << std::endl;
+        if(this->select.length() > 0) {
+            aws.run_s3select(this->select);
+            //std::cout << "res = " << aws.get_result() << std::endl;
+            //this->output2 = aws.get_result();
+
+
+
+            /*std::cout << "chunk = " << this->chunk << std::endl;
+            std::cout << "end ================================================ " << std::endl;*/
+
+            /*std::cout << "aws get result = " << aws.get_result() << std::endl;
+            std::cout << "end ================================================ " << std::endl;*/
+
+            std::cout.flush();
+        }
+
+
+	    /*std::cout << "select res = " << this->output << std::endl;
+        std::cout.flush();*/
+	}
+
+    void OnOK() override
+    {
+        //DBG1("FS::FSWorker::OnOK: output " << output);
+        Napi::Env env = Env();
+        Napi::Value napi_res = Napi::Buffer<char>::/*New*/Copy(env, chunk.data(), chunk.length());
+        this->deferred->Resolve(napi_res);
+    }
+
+    void OnError(Napi::Error const& error) override
+    {
+        Napi::Env env = Env();
+        //DBG1("FS::FSWorker::OnError: " << _desc << " " << DVAL(error.Message()));
+        //ReportWorkerStats(1);
+        auto obj = error.Value();
+        /*if (_errno) {
+            obj.Set("code", Napi::String::New(env, uv_err_name(uv_translate_sys_error(_errno))));
+        }*/
+        //_deferred.Reject(obj);
+    }
+};
+
+class CephS3lectNapi : public Napi::ObjectWrap<CephS3lectNapi> {
+
+        public:
+            static Napi::Object Init(Napi::Env env, Napi::Object exports);
+            CephS3lectNapi(const Napi::CallbackInfo& info);
+            Napi::Value Write(const Napi::CallbackInfo& info);
+            Napi::Value Flush(const Napi::CallbackInfo& info);
+            ~CephS3lectNapi();
+
+        private:
+            static Napi::FunctionReference constructor;
+            InputFormat input_format;
+            s3selectEngine::s3select s3select;
+            s3selectEngine::csv_object *csv_object = NULL;
+            s3selectEngine::json_object *json_object = NULL;
+            Napi::Promise::Deferred *deferred = NULL;
+		    CephSelectWorker worker;
```

Comment:
indentation. you should use the formatting rules as defined in the repo -
https://github.com/noobaa/noobaa-core/blob/master/.clang-format

I use the vscode c++ extension to format native code, and it reads that config file.